src/Features/Core/Portable/CodeLens/IDisplayInfoLanguageServices.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.CodeLens
+{
+    internal interface IDisplayInfoLanguageServices : ILanguageService
+    {
+        /// <summary>
+        /// Indicates if the given node is a declaration of some kind of symbol. 
+        /// For example a class for a method declaration.
+        /// </summary>
+        bool IsDeclaration(SyntaxNode node);
+
+        /// <summary>
+        /// Indicates if the given node is a namespace import.
+        /// </summary>
+        bool IsDirectiveOrImport(SyntaxNode node);
+
+        /// <summary>
+        /// Indicates if the given node is an assembly level attribute "[assembly: MyAttribute]"
+        /// </summary>
+        bool IsGlobalAttribute(SyntaxNode node);
+
+        /// <summary>
+        /// Indicates if given node is DocumentationCommentTriviaSyntax
+        /// </summary>
+        bool IsDocumentationComment(SyntaxNode node);
+
+        /// <summary>
+        /// Gets the node used for display info
+        /// </summary>
+        SyntaxNode GetDisplayNode(SyntaxNode node);
```

Comment:
GetDisplayXXX seems only method needs to be here. other ones seem should be move to SyntaxFactsService